tests/api2/test_alert.py

```diff
@@ -0,0 +1,75 @@
+from time import sleep
+
+import pytest
+
+from auto_config import pool_name
+from middlewared.test.integration.utils import call, ssh
+
+
+ID_PATH = "/dev/disk/by-partuuid/"
+
+
+def get_alert_by_id(alert_id):
+    return next(filter(lambda alert: alert["id"] == alert_id, call("alert.list")), None)
+
+
+def wait_for_alert(timeout=120):
+    for _ in range(timeout):
+        for alert in call("alert.list"):
+            if (
+                alert["source"] == "VolumeStatus" and
+                alert["args"]["volume"] == pool_name and
+                alert["args"]["state"] == "DEGRADED"
+            ):
+                return alert["id"]
+        sleep(1)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def degraded_pool_gptid():
+    get_pool = call("pool.query", [["name", "=", pool_name]], {"get": True})
+    gptid = get_pool["topology"]["data"][0]["path"].replace(ID_PATH, "")
+    ssh(f"zinject -d {gptid} -A fault {pool_name}")
+    return gptid
+
+
+@pytest.fixture(scope="module")
+def alert_id(request, degraded_pool_gptid):
+    call("alert.process_alerts")
+    timeout = request.param if hasattr(request, "param") else 120
```

Comment:
This allows a timeout to be passed to the fixture since `wait_for_alert` accepts the optional argument. So currently, nothing

Reply:
What's the use case for this?

Reply:
To be honest, [this commit](https://github.com/truenas/middleware/pull/14415/commits/8b7ffe067919cf654484f30851fcf0fbc13bf175) was experimental. Wanted to figure out how to do something like this and it worked so I left it in. Will remove :)

Reply:
But to answer your question
```python
@pytest.mark.parametrize('alert_id', [60], indirect=True)
def test_some_alert_functionality(alert_id): ...
```
in case 120 was too long for whatever reason